cpp/src/reductions/group_std.cu

```diff
@@ -0,0 +1,156 @@
+                            thrust::make_counting_iterator(0),
+                            [=] __device__ (gdf_size_type i) -> T {
+                              return (values_valid and not bit_mask::is_valid(values_valid, i))
+                                     ? 0 : values_data[i];
+                            }),
+                          thrust::make_discard_iterator(),
+                          sums.begin());
+    print(sums, "sums");
+
+    // TODO: use target_type for sums and result_data
+    T* d_sums = sums.data().get();
+
+    auto values_it = thrust::make_transform_iterator(
+      thrust::make_counting_iterator(0),
+      [=] __device__ (gdf_size_type i) {
+        if (values_valid and not bit_mask::is_valid(values_valid, i))
+          return 0.0;
+        
+        double x = values_data[i];
+        gdf_size_type group_idx = d_group_labels[i];
+        gdf_size_type group_size = d_group_sizes[group_idx];
+        
+        // prevent divide by zero error
+        if (group_size == 0 or group_size - ddof <= 0)
+          return 0.0;
+
+        double mean = static_cast<double>(d_sums[group_idx])/group_size;
+        return (x - mean) * (x - mean) / (group_size - ddof);
+      }
+    );
+
+    thrust::reduce_by_key(rmm::exec_policy(stream)->on(stream),
+                          group_labels.begin(), group_labels.end(), values_it, 
+                          thrust::make_discard_iterator(),
+                          result_data);
+
+    // set nulls
+    if (result_valid) {
+      thrust::for_each_n(thrust::make_counting_iterator(0), group_sizes.size(),
+        [=] __device__ (gdf_size_type i){
+          gdf_size_type group_size = d_group_sizes[i];
+          if (group_size == 0 or group_size - ddof <= 0)
+            bit_mask::clear_bit_safe(result_valid, i);
+          else
+            bit_mask::set_bit_safe(result_valid, i);
+        });
+    }
+    
+  }
+
+  template <typename T, typename... Args>
+  std::enable_if_t<!std::is_arithmetic<T>::value, void >
+  operator()(Args&&... args) {
+    CUDF_FAIL("Only numeric types are supported in variance");
+  }
+};
+
+} // namespace anonymous
+
+
+namespace cudf {
+namespace detail {
+
+void group_var(gdf_column const& values,
+               rmm::device_vector<gdf_size_type> const& group_labels,
+               rmm::device_vector<gdf_size_type> const& group_sizes,
+               gdf_column * result,
+               gdf_size_type ddof,
+               cudaStream_t stream)
+{
+  type_dispatcher(values.dtype, var_functor{},
+    values, group_labels, group_sizes, result, ddof, stream);
+}
+
+void group_std(gdf_column const& values,
+               rmm::device_vector<gdf_size_type> const& group_labels,
+               rmm::device_vector<gdf_size_type> const& group_sizes,
+               gdf_column * result,
+               gdf_size_type ddof,
+               cudaStream_t stream)
+{
+  type_dispatcher(values.dtype, var_functor{},
```

Comment:
std is just calling var? Is this just a placeholder?

Reply:
Yes, this is a placeholder. There needs to be a vectorized `sqrt` right after this.